### DISCUSSION TITLE
ScriptReference/Vector3.SignedAngle

### DISCUSSION THREAD

**Discussion by docscomments** [Developer-Notes-Extension/Unity-Notes#284] [Notes]
# ScriptReference/Vector3.SignedAngle

https://docs.unity3d.com/ScriptReference/Vector3.SignedAngle.html

**Comment by docscomments** [comment]
**Comment by `@HilariousCow`:**  

Gotcha:  
Do not expect the "from" and "to" vectors to be flattened against the Axis when calculating the angle. If the inputs are pointing at all up or down in the plane defined by the axis, you're going to get the "Angle between two 3D vectors" returned. NOT the "Angle between two 3D vectors flattened on a plane".  
Workaround: Make sure that you flatten the "from" and "to" vectors against the axis: i.e.  
  

```csharp
Vector3.SignedAngle(Vector3.ProjectOnPlane(from, Vector3.up), Vector3.ProjectOnPlane(to, Vector3.up), Vector3.up);
```

**Reply by ThePhiMa** [reply]
Thank you for your advice, your solution resolved a bug I had for multiple days now :)
